js/src/singletons/traceable.ts

```diff
@@ -45,20 +45,22 @@ export const AsyncLocalStorageProviderSingleton =
  *
  * @returns The run tree for the given context.
  */
-export const getCurrentRunTree = () => {
+export function getCurrentRunTree(): RunTree;
+
+export function getCurrentRunTree(throwIfMissing: true): RunTree;
+
+export function getCurrentRunTree(throwIfMissing: boolean): RunTree | undefined;
+
+export function getCurrentRunTree(throwIfMissing = true) {
   const runTree = AsyncLocalStorageProviderSingleton.getInstance().getStore();
-  if (!isRunTree(runTree)) {
+  if (throwIfMissing && !isRunTree(runTree)) {
```

Comment:
nit: can we prefer falsey defaults? Not a huge deal in this case but better practice

```
export function getCurrentRunTree(permitAbsentRunTree: boolean) {
```